controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1164,6 +1165,19 @@ func (r *BareMetalHostReconciler) actionPreparing(prov provisioner.Provisioner,
 		return recordActionFailure(info, metal3api.PreparationError, provResult.ErrorMessage)
 	}
 
+	if hfcDirty && started {
+		// handle updating hostfirmwarecomponent
+		info.log.Info("saving host firmware components")
+		_, err := r.saveHostFirmwareComponents(prov, info, hfc)
```

Comment:
Why are we ignoring the return value?

Reply:
it's returning a dirty flag, thinking a bit now maybe I don't need it.. what do you think?

Reply:
If it's always going to be true (because we already checked `hfcDirty`), we don't need a flag.
If it isn't, we should look at the return value.

Reply:
oh, I think I understood now, in case we return false and no error we should  send an `actionContinue`?

Reply:
More importantly, you don't need to call Update if dirty is false. And yes, after that you should probably return actionContinue.

Reply:
Tks Dmitry!

Reply:
.. but you cannot just return actionContinue here because then you'll never reach line 1181. So line 1194 probably already does the right return.

Reply:
I've changed to

		hfcStillDirty, err := r.saveHostFirmwareComponents(prov, info, hfc)
		if err != nil {
			return actionError{errors.Wrap(err, "could not save the host firmware components")}
		}

		if hfcStillDirty {
			info.log.Info("going to update the host firmware components")
			if err := r.Status().Update(info.ctx, hfc); err != nil {
				return actionError{errors.Wrap(err, "failed to update hostfirmwarecomponents status")}
			}
		}